presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/ExtendedHiveMetastore.java

```diff
@@ -82,6 +82,14 @@
             String tableName,
             Map<Column, Domain> partitionPredicates);
 
+    default List<PartitionNameWithVersion> getPartitionNamesWithVersionByFilter(
```

Comment:
For `ExtendedHiveMetastore`, we might be able to make it non-default and actually implement them (throw unsupported) in the implementations?